packages/colorpickers/src/elements/ColorPicker/index.tsx

```diff
@@ -0,0 +1,473 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, {
+  useState,
+  useEffect,
+  useCallback,
+  useReducer,
+  forwardRef,
+  ChangeEvent,
+  HTMLAttributes
+} from 'react';
+import PropTypes from 'prop-types';
+import { Label } from '@zendeskgarden/react-forms';
+import { ColorWell } from './ColorWell';
+import { isValidHex } from '../../utils/validation';
+import { hsvToHsl, hslToRgb, rgbToHsl } from '../../utils/conversion';
+import {
+  StyledHue,
+  StyledSliderGroup,
+  StyledLabel,
+  StyledInput,
+  StyledAlpha,
+  StyledSliders,
+  StyledHexField,
+  StyledRGBAField,
+  StyledInputGroup,
+  StyledPreview,
+  StyledColorPicker,
+  StyledCheckered,
+  StyledAlphaGradient,
+  StyledAlphaField,
+  StyledHueField
+} from '../../styled';
+import { toState, reducer } from './reducer';
+import { IColor, IHSVColor } from '../../utils/types';
+import { parseToRgb, rgb as rgbToString } from 'polished';
+export interface IColorPickerLabels {
+  hueSlider?: string;
+  alphaSlider?: string;
+  hex?: string;
+  red?: string;
+  green?: string;
+  blue?: string;
+  alpha?: string;
+}
+
+export interface IColorPickerProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, 'color' | 'onChange'> {
+  /** A default color for a uncontrolled color picker. Can be a valid color string or IColor state */
+  defaultColor?: string | IColor;
+  /** A color value for a controlled color picker. Can be a valid color string or IColor state */
+  color?: string | IColor;
+  /**
+   * Handles color picker changes
+   *
+   * @param {Object} color a color picker state
+   */
+  onChange?: (color: IColor) => void;
+  /** Replaces the default labels within the color picker */
+  labels?: IColorPickerLabels;
+  /** Autofocuses the hex input element */
+  autofocus?: boolean;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLDivElement>
+ */
+export const ColorPicker = forwardRef<HTMLDivElement, IColorPickerProps>(
+  ({ color, defaultColor, labels = {}, autofocus, onChange, ...props }, ref) => {
+    const isControlled = color !== null && color !== undefined;
+    const [uncontrolledState, dispatch] = useReducer(reducer, toState(defaultColor));
+
+    useEffect(() => {
+      if (isControlled === false) {
+        onChange && onChange(uncontrolledState);
+      }
+    }, [uncontrolledState, onChange, isControlled]);
```

Comment:
Could you go into the reasoning behind having both a controlled vs uncontrolled state manager here?

It looks like the state objects below replicate what is already being accomplished in the reducer.